### PR TITLE
Retry policy for SignalR transport protocol

### PR DESCRIPTION
**Pull request by thomashilzendegen**
Closing #273

### CODE REVIEW

File: src/Thinktecture.Relay.Connector.Protocols.SignalR/DiscoveryDocumentRetryPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		/// <param name="relayConnectorOptions">An <see cref="IOptions{TOptions}"/>.</param>
+		public DiscoveryDocumentRetryPolicy(ILogger<DiscoveryDocumentRetryPolicy> logger,
+			IOptions<RelayConnectorOptions> relayConnectorOptions)
+		{
+			_logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+			if (relayConnectorOptions == null) throw new ArgumentNullException(nameof(relayConnectorOptions));
+
+			_minimumDelay = (int)relayConnectorOptions.Value.DiscoveryDocument.ReconnectMinimumDelay.TotalSeconds;
+			_maximumDelay = (int)(relayConnectorOptions.Value.DiscoveryDocument.ReconnectMaximumDelay.TotalSeconds + 1);
```
Comment (gingters) [inline src/Thinktecture.Relay.Connector.Protocols.SignalR/DiscoveryDocumentRetryPolicy.cs:30]: Why +1 second?

Comment (thomashilzendegen) [reply src/Thinktecture.Relay.Connector.Protocols.SignalR/DiscoveryDocumentRetryPolicy.cs:30]: It's an exclusive upper bound.

Comment (thomashilzendegen) [reply src/Thinktecture.Relay.Connector.Protocols.SignalR/DiscoveryDocumentRetryPolicy.cs]: I added a comment to explain this for future "questioning".

Comment (gingters) [reply src/Thinktecture.Relay.Connector.Protocols.SignalR/DiscoveryDocumentRetryPolicy.cs]: Ahh. Makes sense ;)

**Review by gingters — APPROVED** [review]

**Pull request merged**